spring-kafka/src/main/java/org/springframework/kafka/listener/FailedBatchProcessor.java

```
@@ -254,35 +246,33 @@ private <K, V> ConsumerRecords<K, V> seekOrRecover(Exception thrownException, @N
 			return ConsumerRecords.empty();
 		}
 		else {
-			if (indexArg == 0) {
-				return (ConsumerRecords<K, V>) data; // first record just rerun the whole thing
-			}
-			else {
-				try {
-					if (getFailureTracker().recovered(remaining.get(0), thrownException, container,
-							consumer)) {
-						remaining.remove(0);
-					}
-				}
-				catch (Exception e) {
+			try {
+				if (getFailureTracker().recovered(remaining.get(0), thrownException, container,
+						consumer)) {
+					remaining.remove(0);
 				}
-				Map<TopicPartition, List<ConsumerRecord<K, V>>> remains = new HashMap<>();
-				remaining.forEach(rec -> remains.computeIfAbsent(new TopicPartition(rec.topic(), rec.partition()),
-						tp -> new ArrayList<ConsumerRecord<K, V>>()).add((ConsumerRecord<K, V>) rec));
-				return new ConsumerRecords<>(remains);
 			}
+			catch (Exception e) {
+			}
```

Empty catch - can we address this?

---

fixed